Introduce and implement ReqViewChange message type

sergefdrv opened the pull request:
This pull request introduces and implements ReqViewChange message type. This message will be generated and multicast by a replica upon request timeout expiration. Please refer to individual commit messages for more detailed description.

NB. This draft pull request borrows commits from #155 and to be rebased after #155 is merged.

nhoriguchi reviewed (approved):
Looks good to me. Please rebase onto current master branch.

Commit:
messages: Introduce ReqViewChange message interface

The new interface represents REQ-VIEW-CHANGE protocol message type.
The interface slightly differs from the way REQ-VIEW-CHANGE message is
described in the protocol paper. Namely, the interface does not have a
method for "current view number". The main reason for that change is
as follows.

Different replicas might not update their local current view numbers
synchronously. Specifically, even if all replicas start view change
from the same current view number v, it may happen that some replicas
successfully transition into the new view v+1, whereas other replicas
may receive message from the new primary late and trigger another view
change first. In that case, the former replicas will eventually
request view change to v+2 with local current view number v+1, whereas
the latter will still use view number v.

Therefore, it would not be correct to include "current view number"
field of REQ-VIEW-CHANGE message for matching messages received from
different replicas. Doing so would break liveness guarantee in the
aforementioned scenario, since there might not be enough correct
replicas having the same local current view number. On the other hand,
there is no hint for any possible usage of the field in the paper.

Signed-off-by: Sergey Fedorov <sergey.fedorov@neclab.eu>

Commit:
messages/protobuf: Implement ReqViewChange message interface

Signed-off-by: Sergey Fedorov <sergey.fedorov@neclab.eu>

sergefdrv commented:
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/6ded82b1e6b40efd465abf2897e44a15d764f3c0..cd70a84bfd2ef08b52903e7a97e4cef264e94889):

- rebase with conflict resolution
- require `ReqViewChange` to statisfy `PeerMessage` interface
